include/arch/arm/aarch64/exc.h

```diff
@@ -26,7 +26,10 @@ extern "C" {
 
 struct __esf {
 	struct __basic_sf {
-		u64_t regs[20];
+		u64_t spsr;
+		u64_t elr;
+		u64_t pc;
+		u64_t regs[3];
```

Comment:
I am not sure about this change.

`__esf`/`__basic_sf`, being a "stack frame", is supposed to match the structure of the stack frame saved during exception/interrupt entry (i.e. the structure of data saved in the stack by `z_arm64_enter_exc`). That does not seem to be the case here.

Reply:
this is not the case for AArch64 and ESF is a misnomer for ARM64. This struct is only used by new thread to correctly set the initial stack, nothing else. After the thread is created and running this struct is never actually used.

Reply:
Probably it's better to rename it. ARC for example uses `init_stack_frame` that could be a better choice.

Reply:
the real problem here is that `typedef struct __esf z_arch_esf_t;` and that's clearly wrong. I'll fix that.